src/daft-core/src/array/ops/cast.rs

```diff
@@ -2352,6 +2352,9 @@ mod tests {
 
         let scale: usize = rng.gen_range(0..=MAX_SCALE);
```

Comment:
would it simpler to set the lower bound for scale to 1 instead?

Reply:
IMO keeping the lower bound of the scale to 0 is fine.

I would say instead of doing `EPSILON * factor` below, simply doing
```
let epsilon = if scale == 0 { 1.0 } else { 0.1 };
```

Reply:
Thanks! Looks good